data_pipeline_api/java_api/src/main/java/uk/ramp/metadata/MetadataItem.java

```diff
@@ -0,0 +1,108 @@
+package uk.ramp.metadata;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.immutables.value.Value.Immutable;
+import uk.ramp.config.Config.OverrideItem;
+
+@Immutable
+@JsonSerialize
+@JsonDeserialize
+public interface MetadataItem {
+  Optional<String> filename();
+
+  Optional<String> version();
+
+  Optional<String> extension();
+
+  Optional<String> component();
+
+  @JsonProperty("data_product")
+  Optional<String> dataProduct();
+
+  @JsonProperty("run_id")
+  Optional<String> runId();
+
+  @JsonProperty("verified_hash")
+  Optional<String> verifiedHash();
+
+  Optional<String> calculatedHash();
+
+  Optional<String> source();
+
+  default boolean isSuperSetOf(MetadataItem key) {
+    return key.filename().map(k -> k.equals(filename().orElse(""))).orElse(true)
```

Comment:
You could a method here, not sure what to call it, to do:
```
boolean keyIsEitherNotPresentOrEqual(Optional<String> key, Optional<String> other) {
    return key.map(k -> k.equals(superset.orElse(""))).orElse(true);
}
```

Reply:
Updated this, hopefully is more clear now. I generally try to avoid passing in Optionals through parameters as it is considered bad practice by some. So I have passed in empty string instead, doing something quite similar to your suggestion. Let me know what you think.